modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/TokenImageFetcher.swift

```diff
@@ -130,6 +136,10 @@ public class TokenImageFetcherImpl: TokenImageFetcher {
                                           blockChainNameColor: UIColor,
                                           serverIconImage: UIImage?) -> TokenImage? {
 
+        if tokenGroupsIdentifier.isSpam(address: contractAddress.eip55String, chainID: server.chainID) {
+            return .init(image: .image(.loaded(image: spamImage)), isFinal: true, overlayServerIcon: nil)
```

Comment:
Didn't test. But I think `overlayServerIcon: ` still has to be either `nil` or `not-nil` using the `switch-case` a few lines later. That's how the current hardcoded spam (or any token image) loaded from the `iconassets` repo are displayed, right?

Reply:
Updated. Please check @hboon